esy/bin/esyCommand.ml

```diff
@@ -1190,8 +1190,11 @@ let getSandboxSolution installSandbox =
   return solution
 
 let solve {CommonOptions. installSandbox; _} () =
+  let open EsyInstall in
   let open RunAsync.Syntax in
   let%bind _ : Solution.t = getSandboxSolution installSandbox in
+  let unused = Resolver.getUnusedResolutions installSandbox.resolver in
+  let%lwt () = Lwt_list.iter_p (fun name -> (Logs_lwt.warn (fun m -> m "Resolution '%s' is unused" name))) unused in
```

Comment:
I'd suggest to tweak wording a bit (for consistency with other messages) and add info about the manifest in which resolutions are defined:
```suggestion
  let%lwt () =
    let log resolution =
      Logs_lwt.warn (
        fun m ->
          m "resolution %a is unused (defined in %a)"
          Fmt.(quote string)
          resolution
          ManifestSpec.pp
          installSandbox.spec.manifest
      )
    in
    Lwt_list.iter_s log unused
  in
```

The result would look like this:

<img width="494" alt="screen shot 2018-11-13 at 12 37 25" src="https://user-images.githubusercontent.com/30594/48404254-ec1c4900-e740-11e8-88cc-6ef27671b736.png">